core/src/test/java/org/elasticsearch/common/StringsTests.java

```diff
@@ -19,6 +19,8 @@
 
 package org.elasticsearch.common;
 
+import com.carrotsearch.randomizedtesting.generators.CodepointSetGenerator;
+
 import org.elasticsearch.test.ESTestCase;
 
 public class StringsTests extends ESTestCase {
```

Comment:
Wherever these end up, I think that there should be a test for the null string and the empty string being valid header values because those are edge cases that are easily messed up in some refactoring.